windows-installer/stackdriver_logging_agent.nsi

```diff
@@ -0,0 +1,311 @@
+; GLOBAL VARIABLES
+;--------------------------------
+
+; Company or Publisher name.
+!define COMPANY "Stackdriver"
+
+; Software name used for display to users.
+!define DISPLAY_NAME "${COMPANY} Logging Agent"
+
+; Software name with no white space.
+!define COMPRESSED_NAME "${COMPANY}LoggingAgent"
+
+; Registry entry key to store arbitrary information.
+!define REG_KEY "Software\${COMPANY}\LoggingAgent"
+
+; Uninstaller registry key, used to register the logging agent.
+!define STACKDRIVER_LOGGING_UNINST_REG_KEY "${UNINST_REG_KEY}\${COMPRESSED_NAME}"
+
+; Directory used to install all files into.  This makes uninstall easier and safer.
+!define MAIN_INSTDIR "$INSTDIR\Main"
+
+; Absolute location of the uninstaller.
+!define UNINSTALLER_LOCATION "$INSTDIR\uninstall.exe"
+
+; The name of fluentd config template file, this is bundled into the script.
+!define FLUENTD_CONFIG_TEMPLATE "fluent-template.conf"
+
+; Absolute location of the fluentd config file.
+!define FLUENTD_CONFIG_LOCATION "$INSTDIR\fluent.conf"
+
+; Name of the main zip file, this is bundled into the script.
+!define ZIP_FILE "${COMPRESSED_NAME}.zip"
+
+
+;--------------------------------
+; GENERAL CONFIGURATION 
+;--------------------------------
+
+; Needed to properly display all languages, may not work on older windows OSs.
+Unicode true
+
+; Name used for display throughout the installer.
+Name "${DISPLAY_NAME}"
+
+; Output location/name of the installer executable.
+OutFile "${COMPRESSED_NAME}_unsigned.exe"
+
+; Require admin level logs access, this is required as we need to read event logs.
+RequestExecutionLevel admin
+
+
+;--------------------------------
+; INCLUDES
+;--------------------------------
+
+; General includes.
+!include "FileFunc.nsh" ; Needed for GetSize
+!include "StrFunc.nsh" ; Needed for StrTrimNewLines
+!include "WordFunc.nsh" ; Needed for WordFind
+
+; Stackdriver includes
+!include "stackdriver_language_util.nsh"
+!include "stackdriver_ui.nsh" 
+!include "stackdriver_util.nsh"
+
+
+;--------------------------------
+; FUNCTIONS
+;--------------------------------
+
+; Define StrTrimNewLines function to use below.
+${StrTrimNewLines}
+
+
+;--------------------------------
+; GENERAL UI SET UP
+;--------------------------------
+
+; Configures auto save for the users selected language. See stackdriver_language_util.nsh.
+!insertmacro STACKDRIVER_SAVE_USER_LANGUAGE "${REG_KEY}"
+
+; Configures the UI and pages for the installer. See stackdriver_ui.nsh.
+!insertmacro STACKDRIVER_UI "true"
+
+; Adds all supported languages. See stackdriver_language_util.nsh.
+!insertmacro STACKDRIVER_INCLUDE_ALL_LANGUAGES
+
+
+;--------------------------------
+; INSTALLER FUNCTIONS
+;--------------------------------
+
+; Function called as soon as the installer is initialized, before the GUI.
+Function .onInit
+  ; Set the install directory, we cannot do this at compile time as we are
+  ; using a runtime variable.
+  ReadEnvStr $0 SYSTEMDRIVE 
```

Comment:
Can you please get rid of the trailing whitespace (lines 44, 71, 105, 106, 129, 141, 169, and 189)?